pkgs/development/python-modules/triton/default.nix

```diff
@@ -1,106 +1,56 @@
+    owner = "triton-lang";
+    repo = "triton";
+    # latest branch commit from https://github.com/triton-lang/triton/commits/release/3.0.x/
+    rev = "91f24d87e50cb748b121a6c24e65a01187699c22";
+    hash = "sha256-L5KqiR+TgSyKjEBlkE0yOU1pemMHFk2PhEmxLdbbxUU=";
   };
 
-  patches =
-    [
-      # fix overflow error
-      (fetchpatch {
-        url = "https://github.com/openai/triton/commit/52c146f66b79b6079bcd28c55312fc6ea1852519.patch";
-        hash = "sha256-098/TCQrzvrBAbQiaVGCMaF3o5Yc3yWDxzwSkzIuAtY=";
-      })
-
-      # Upstream startded pinning CUDA version and falling back to downloading from Conda
-      # in https://github.com/triton-lang/triton/pull/1574/files#diff-eb8b42d9346d0a5d371facf21a8bfa2d16fb49e213ae7c21f03863accebe0fcfR120-R123
-      ./0000-dont-download-ptxas.patch
-    ]
-    ++ lib.optionals (!cudaSupport) [
-      # triton wants to get ptxas version even if ptxas is not
-      # used, resulting in ptxas not found error.
-      ./0001-ptxas-disable-version-key-for-non-cuda-targets.patch
-    ];
+  # triton wants to download every dependency, even if we are not using cuda.
+  patches = lib.optionals (!cudaSupport) [ ./0000-dont-download-ptxas.patch ];
 
   postPatch =
-    let
-      quote = x: ''"${x}"'';
-      subs.ldFlags =
-        let
-          # Bash was getting weird without linting,
-          # but basically upstream contains [cc, ..., "-lcuda", ...]
-          # and we replace it with [..., "-lcuda", "-L/run/opengl-driver/lib", "-L$stubs", ...]
-          old = [ "-lcuda" ];
-          new = [
-            "-lcuda"
-            "-L${addDriverRunpath.driverLink}"
-            "-L${cudaPackages.cuda_cudart}/lib/stubs/"
-          ];
-        in
-        {
-          oldStr = lib.concatMapStringsSep ", " quote old;
-          newStr = lib.concatMapStringsSep ", " quote new;
-        };
-    in
     ''
       # Use our `cmakeFlags` instead and avoid downloading dependencies
+      # remove any downloads
       substituteInPlace python/setup.py \
-        --replace "= get_thirdparty_packages(triton_cache_path)" "= os.environ[\"cmakeFlags\"].split()"
-
-      # Already defined in llvm, when built with -DLLVM_INSTALL_UTILS
-      substituteInPlace bin/CMakeLists.txt \
-        --replace "add_subdirectory(FileCheck)" ""
+        --replace-fail "get_json_package_info(), get_pybind11_package_info()" ""\
+        --replace-fail "get_pybind11_package_info(), get_llvm_package_info()" ""\
+        --replace-fail 'packages += ["triton/profiler"]' ""\
+        --replace-fail "curr_version != version" "False"
 
       # Don't fetch googletest
       substituteInPlace unittest/CMakeLists.txt \
-        --replace "include (\''${CMAKE_CURRENT_SOURCE_DIR}/googletest.cmake)" ""\
-        --replace "include(GoogleTest)" "find_package(GTest REQUIRED)"
-
-      cat << \EOF >> python/triton/common/build.py
-      def libcuda_dirs():
-          return [ "${addDriverRunpath.driverLink}/lib" ]
-      EOF
```

Comment:
You're removing the `/run/opengl-driver/lib` integration from postPatch but not adding any replacement?

Reply:
I have to check that too. Is your testing script working. How would you test it?

Reply:
This means spending time on torch but you can `nix run` the `python3Packages.torch.tests.tester-compileCuda` (built with `config.cudaSupport = true`)

Reply:
But this is going to fail without updating the libcuda discovery logic. Maybe https://github.com/NixOS/nixpkgs/pull/292996/files#diff-12d5872269fe05b4b44d02433ddaf5804f72b72ccda1c3b488d50684ba3cf889R106-R107 is still applicable, otherwise one just has to grep the upstream source for mentions of libcuda

Reply:
I wasn't exactly right:

- `tester-compileCuda` fails regardless of cuda because there's no 3.12 support in their jit thingy (dynamo)
- `tester-cudaAvailable` succeeds regardless of whether triton supports cuda

We need a separate runtime test for triton...